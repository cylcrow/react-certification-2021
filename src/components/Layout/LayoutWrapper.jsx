import React from 'react';
import Navbar from './Navbar';
import MainContent from './MainContent';

const LayoutWrapper = () => {
  return (
    <div data-testid="layout-wrapper">
      <Navbar />
      <MainContent />
    </div>
  );
};

export default LayoutWrapper;
