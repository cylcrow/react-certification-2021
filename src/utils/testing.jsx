import React from "react";
import { render } from '@testing-library/react';
import { ThemeProvider } from "styled-components";
import { lightTheme } from '../providers/theme/themes';

export const renderWithTheme = (Component, theme = lightTheme) => {
    return render(
        <ThemeProvider theme={{ theme, switchTheme: jest.fn() }}>
        { Component }
        </ThemeProvider>
    );
}