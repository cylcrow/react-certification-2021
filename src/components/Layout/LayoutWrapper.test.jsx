import React from 'react';
import '../../utils/testing';
import { render, unmountComponentAtNode } from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { getByTestId } from '@testing-library/react';
import LayoutWrapper from './LayoutWrapper';

let container;
const build = () => {
  render(<LayoutWrapper />, container);
  return {
    Navbar: () => getByTestId(container, 'navbar'),
    MainContent: () => getByTestId(container, 'main-content'),
  };
};

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  return container;
});

describe('LayoutWrapper', () => {
  it('renders', () => {
    const wrapper = shallow(<LayoutWrapper />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('displays Navbar and MainContent', () => {
    const { Navbar, MainContent } = build();

    expect(Navbar()).toBeInTheDocument();
    expect(MainContent()).toBeInTheDocument();
  });
});
