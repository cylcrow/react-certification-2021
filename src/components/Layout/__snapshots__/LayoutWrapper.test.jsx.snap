// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutWrapper renders 1`] = `
<div
  data-testid="layout-wrapper"
>
  <Navbar />
  <MainContent />
</div>
`;
